### PR TITLE
Convert serially to save memory

### PR DESCRIPTION
**Pull request by MichaelMa2014**
[Here is my rationale.](https://github.com/tensorflow/magenta/issues/888) I hope it makes sense~ 😄

### CODE REVIEW

**Comment by cghawthorne** [comment]
This seems reasonable to me. I know we've gotten a number of complaints about this script using up too much memory.

Another option would be to not let the queue of items to convert grow beyond a certain size. This would retain whatever benefits there are to multithreading without causing memory issues.

@adarob since you wrote the threaded version, what do you think?

**Comment by Calvin-Shen** [comment]
It seems really solving the problem that memory can not be released in time when generating .tfrecord file from midi files. thanks a lot @MichaelMa2014 @cghawthorne

File: magenta/scripts/convert_dir_to_note_sequences.py [hunk 1]
```diff
@@ -51,14 +52,14 @@
                            'DEBUG, INFO, WARN, ERROR, or FATAL.')
 
 
-def queue_conversions(root_dir, sub_dir, pool, recursive=False):
-  """Queues callables for converting files to NoteSequences via threadpool.
+def queue_conversions(root_dir, sub_dir, writer, recursive=False):
```
Comment (adarob) [inline magenta/scripts/convert_dir_to_note_sequences.py]: Rename to something like "convert_files"

**Comment by Calvin-Shen** [comment]
it still exists some proble when converting midi files to .tfrecord file. it seems having no memory problem ,but it still killed my progress with not converting complete files.  I don't know whtat causes this problem. @MichaelMa2014 @cghawthorne @adarob

**Comment by MichaelMa2014** [comment]
@Calvin-Shen Please use `sudo dmesg` to check why your OS killed your process.

**Comment by Calvin-Shen** [comment]
I just do with "sudo dmesg" in terminal.and showing as follows:
wechatimg2111508138832_ pic

**Comment by Calvin-Shen** [comment]
I just use your updated code. I hope you help me to solve the problem.  Thank you~ @adarob @MichaelMa2014 @cghawthorne

**Comment by MichaelMa2014** [comment]
Yes it seems this is a memory issue. Could you please try using only part of your dataset and see if the issue still exists? Sometimes one corrupted file can eat up all the memory and it happened to me before.

**Comment by MichaelMa2014** [comment]
@Calvin-Shen Here is an example of a corrupted midi file. It drained all my memory.
```
>>> import pretty_midi as pm
>>> pm.PrettyMIDI("30073.mid")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/v-chenxm/.local/lib/python3.5/site-packages/pretty_midi/pretty_midi.py", line 84, in __init__
    ' it is likely corrupt'.format(max_tick)))
ValueError: MIDI file has a largest tick of 4295008119, it is likely corrupt
```

**Comment by Calvin-Shen** [comment]
@MichaelMa2014 ok~thanks for your patient answer！

**Pull request merged**